core/arch/arm/kernel/thread_spmc.c

```diff
@@ -1898,8 +1928,32 @@ struct mobj_ffa *thread_spmc_populate_mobj_from_rx(uint64_t cookie)
 	if (!mf)
 		goto out;
 
-	if (set_pages(descr->address_range_array,
-		      READ_ONCE(descr->address_range_count), num_pages, mf)) {
+	assert(frag_length <= length && frag_length <= my_rxtx.size);
```

Comment:
`frag_length` and `length` come from a source external to OP-TEE. `assert()` is not enough (stubbed when not in debug build mode). I think you need to treat that as an error case.

Reply:
Done

Reply:
If the latest fix doesn't make sense, let me know, thanks